src/output/plugins/RecorderOutputPlugin.cxx

```diff
@@ -200,9 +268,78 @@ RecorderOutput::Commit()
 		throw;
 	}
 
+	/* move file to archive if requested */
+	if (archive_requested && can_archive) {
+
+		// wait for previous move operation to finish
+		if (archive_thread.IsDefined())
+			archive_thread.Join();
+
+		// run move/copy asynchronously to not block output operation
+		//
+		archive_source = strdup(path.c_str());
+		archive_dest = strdup(archive_path.c_str());
+
+		archive_thread.Start();
+
+		archive_requested = 0;
+	}
+	else
+	{
+	    /* delete file if requested */
+	    if (delete_after_record) {
+		    FormatDebug(recorder_domain, "Deleting \"%s\"", path.c_str());
+		    
+		    unlink(path.c_str());
+	    }
+	}
+
 	delete file;
 }
 
+void
+RecorderOutput::ArchiveTask() {
+	int rc;
```

Comment:
What's the point of this flag? Instead of moving the logger code out of the try/catch, why not log events where they happen?